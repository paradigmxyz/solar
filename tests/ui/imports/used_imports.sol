//@ compile-flags: -Zwarn-unused
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.0;

import "./auxiliary/Library.sol";
import "./auxiliary/Library2.sol" as Lib2;
import {Helper, Utils as U} from "./auxiliary/Helpers.sol";
import * as AllHelpers from "./auxiliary/AllHelpers.sol";

contract Test {
    // Use all imports
    function test() public {
        Library.libFunc();
        Lib2.lib2Func();
        Helper.doSomething();
        U.utilFunc();
        AllHelpers.AllHelper1.help1();
    }
}