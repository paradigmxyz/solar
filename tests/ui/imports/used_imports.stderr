warning: unused import: `U`
  --> ROOT/tests/ui/imports/used_imports.sol:LL:CC
   |
LL | import {Helper, Utils as U} from "./auxiliary/Helpers.sol";
   | -----------------------------------------------------------
   |

warning: unused function: `test`
  --> ROOT/tests/ui/imports/used_imports.sol:LL:CC
   |
LL | /     function test() public {
LL | |         Library.libFunc();
...  |
LL | |         AllHelpers.AllHelper1.help1();
LL | |     }
   | |_____-
   |

